src/vivarium/framework/results/manager.py

```diff
@@ -269,115 +270,65 @@ def _bin_data(data: Union[pd.Series, pd.DataFrame]) -> pd.Series:
+        is_stratified: bool,
+        **kwargs,
+    ):
+        self.logger.debug(f"Registering observation {kwargs['name']}")
+        # Are we stratified or not?
+        if is_stratified:
+            additional_stratifications = kwargs.get("additional_stratifications", [])
+            excluded_stratifications = kwargs.get("excluded_stratifications", [])
+            self._warn_check_stratifications(
+                additional_stratifications, excluded_stratifications
+            )
+            stratifications = self._get_stratifications(
+                kwargs.get("stratifications", []),
+                additional_stratifications,
+                excluded_stratifications,
+            )
+            kwargs["stratifications"] = stratifications
+            del kwargs["additional_stratifications"]
+            del kwargs["excluded_stratifications"]
 
-    def register_adding_observation(
-        self,
-        name: str,
-        pop_filter: str,
-        when: str,
-        requires_columns: List[str],
-        requires_values: List[str],
-        results_formatter: Callable[[str, pd.DataFrame], pd.DataFrame],
-        additional_stratifications: List[str],
-        excluded_stratifications: List[str],
-        aggregator_sources: Optional[List[str]],
-        aggregator: Callable[[pd.DataFrame], Union[float, pd.Series[float]]],
-    ) -> None:
-        self.logger.debug(f"Registering observation {name}")
-        self._warn_check_stratifications(additional_stratifications, excluded_stratifications)
-        self._results_context.register_adding_observation(
-            name=name,
-            pop_filter=pop_filter,
-            when=when,
-            results_formatter=results_formatter,
-            additional_stratifications=additional_stratifications,
-            excluded_stratifications=excluded_stratifications,
-            aggregator_sources=aggregator_sources,
-            aggregator=aggregator,
-        )
+        requires_columns = kwargs.get("requires_columns", [])
+        requires_values = kwargs.get("requires_values", [])
         self._add_resources(requires_columns, SourceType.COLUMN)
         self._add_resources(requires_values, SourceType.VALUE)
+        del kwargs["requires_columns"]
+        del kwargs["requires_values"]
 
-    def register_concatenating_observation(
-        self,
-        name: str,
-        pop_filter: str,
-        when: str,
-        requires_columns: List[str],
-        requires_values: List[str],
-        results_formatter: Callable[[str, pd.DataFrame], pd.DataFrame],
-    ) -> None:
-        self.logger.debug(f"Registering observation {name}")
-        self._results_context.register_concatenating_observation(
-            name=name,
-            pop_filter=pop_filter,
-            when=when,
-            included_columns=["event_time"] + requires_columns + requires_values,
-            results_formatter=results_formatter,
+        if observation_type == ConcatenatingObservation:
```

Comment:
Is this the best way to do this? Alternatively I can add `included_columns` as a BaseObserver attribute that just defaults to an empty list everywhere